### PR TITLE
ESO-494:Adds .md files for External-Secrets Operator

### PR DESCRIPTION
**Pull request by siddhibhor-56**
## Summary by CodeRabbit

* **Documentation**
  * Added new end-to-end repository guides covering operator architecture, Kubernetes API contract conventions (including validation and merge semantics), standardized reconciliation error-handling, and performance/reconciliation best practices.
  * Introduced comprehensive contribution materials, including safe generated-asset update/verification workflows and recommended local development commands.
  * Published testing and integration guidelines, plus consolidated security requirements for hardened deployments and secure metrics/webhook/RBAC behavior.

### CODE REVIEW

File: docs/api-contracts-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+- Immutability on update (initial + updated + expectedError)
+- Boundary values (min/max lengths, min/max items)
+
+Run API tests with `make test-apis` or via the general `make test`.
+
+## Webhook Architecture
+
+This operator does NOT use kubebuilder-generated admission webhooks for its own CRDs. CEL-based CRD validation handles all admission validation. The webhook code in `pkg/controller/external_secrets/` manages the upstream external-secrets operand's `ValidatingWebhookConfiguration` resources (not the operator's own API validation).
+
+## Adding New API Fields Checklist
```
Comment (bharath-b-rh) [inline docs/api-contracts-guidelines.md:165]: Shouldn't we add a section on Godoc too?
```
The Godoc on any field in the API should be sufficiently explained such that an end user understands the purpose, interactions, limitations, and default behavior.
```

File: docs/api-contracts-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+- Top-level `lastTransitionTime` is set on every condition change.
+
+### Status Update Pattern
+All controllers follow the same retry-on-conflict pattern:
+1. Fetch current object from API server.
+2. DeepCopy the changed status into the fetched object.
+3. Call `StatusUpdate` (status subresource update).
+4. Wrap in `retry.RetryOnConflict(retry.DefaultRetry, ...)`.
+
+## Defaults
```
Comment (bharath-b-rh) [inline docs/api-contracts-guidelines.md:107]: Lets add below detail to avoid defining defaults in CRD going forward.

_Excerpt from OpenShift API conventions_
```
With configuration APIs, we typically default fields within the controller and not within the API. This means that the platform has the ability to make changes to the defaults over time.
```

File: docs/api-contracts-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
+# API Contracts Guidelines
```
Comment (bharath-b-rh) [inline docs/api-contracts-guidelines.md:1]: We should also add a rule not define any functions in API package.

File: docs/testing-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+# Testing Guidelines
```
Comment (bharath-b-rh) [inline docs/testing-guidelines.md:1]: This section will need an update when we merge the e2e restructuring PR. Mainly around the labeling of test cases.

**Review by bharath-b-rh — COMMENTED** [review]
Still going through, just a few suggestions for now.

File: docs/api-contracts-guidelines.md (hunk 3 above)
Comment (siddhibhor-56) [reply docs/api-contracts-guidelines.md:1]: Done

File: docs/api-contracts-guidelines.md (hunk 2 above)
Comment (siddhibhor-56) [reply docs/api-contracts-guidelines.md:107]: Done

File: docs/api-contracts-guidelines.md (hunk 1 above)
Comment (siddhibhor-56) [reply docs/api-contracts-guidelines.md:165]: Done

File: docs/testing-guidelines.md (hunk 4 above)
Comment (siddhibhor-56) [reply docs/testing-guidelines.md:1]: True
